[README] Add FAQ about how to make the package load with proprietary deps (#36683)
JuliaRegistries/General: README.md

to wait for a human to review/merge the pull request. You can contact a human
in the `#pkg-registration` channel in the official Julia Slack to expedite this process.

#### My package fails to load because it needs proprietary software/additional setup to work, what can I do?

Before merging a pull request, AutoMerge will check that your package can be installed and
loaded.  It is OK for your package to not be fully functional, but making it at least load
successfully would streamline registration, as it does not require manual intervention from

> the registry maintainers.  This would also let other packages depend on your one, and use

Reviewer: ```suggestion
the registry maintainers.  This would also let other packages depend on it, and use
```

its functionalities only when the proprietary software is available in the system, as done

> for example by the [`CUDA.jl`](https://github.com/JuliaGPU/CUDA.jl) package.  You can at
> least make your package load successfully with AutoMerge by allowing the necessary software
> to be not available when the environment variable `JULIA_REGISTRYCI_AUTOMERGE` is equal to
> `true` (e.g., `get(ENV, "JULIA_REGISTRYCI_AUTOMERGE", "false") == "true"`).

Reviewer: ```suggestion
for example by the [`CUDA.jl`](https://github.com/JuliaGPU/CUDA.jl) package.  If you aren't
able or willing to make your package always loadable without the proprietary dependency
(which is the preferred solution), you can check if the environment variable
`JULIA_REGISTRYCI_AUTOMERGE` is equal to`true` (e.g.,
`get(ENV, "JULIA_REGISTRYCI_AUTOMERGE", "false") == "true"`) and try to make your package
loadable during AutoMerge at least, so that it can be registered without manual intervention.
```

to wait for a human to review/merge the pull request. You can contact a human
in the `#pkg-registration` channel in the official Julia Slack to expedite this process.

#### My package fails to load because it needs proprietary software/additional setup to work, what can I do?

Before merging a pull request, AutoMerge will check that your package can be installed and
loaded.  It is OK for your package to not be fully functional, but making it at least load
successfully would streamline registration, as it does not require manual intervention from
the registry maintainers.  This would also let other packages depend on it, and use
its functionalities only when the proprietary software is available in the system, as done
for example by the [`CUDA.jl`](https://github.com/JuliaGPU/CUDA.jl) package.  If you aren't
able or willing to make your package always loadable without the proprietary dependency
(which is the preferred solution), you can check if the environment variable
`JULIA_REGISTRYCI_AUTOMERGE` is equal to`true` (e.g.,
`get(ENV, "JULIA_REGISTRYCI_AUTOMERGE", "false") == "true"`) and try to make your package

> loadable during AutoMerge at least, so that it can be registered without manual intervention.

Reviewer 2: How about adding something like
````
Examples of packages with proprietary software that use the environment variable
check include:
 - [Gurobi.jl](https://github.com/jump-dev/Gurobi.jl)
 - [CPLEX.jl](https://github.com/jump-dev/CPLEX.jl)
Consult their source code for examples on how to write this check, including as
part of a `Pkg.build` step.
```

Author: Ok, thanks, I mentioned those packages so readers can go and see them, but I didn't mention `Pkg.build` explicitly to not give them too many ideas :wink:
